### PR TITLE
Added check to invalid API endpoints to close issue #3005

### PR DESCRIPTION
**Pull request by TedmanNguyen**
This PR improves accessing URL to return 404 for with invalid endpoint instead of 200.
+ Added handle_invalid_routes (in app.py) and unit test: InvalidResourceTest (in resources_test.py)

**Checks**

- [x] All tests succeed.
- [x] Unit tests added.
- [ ] e2e tests added.
- [ ] Documentation updated.

Closes issue #3005:

### CODE REVIEW

**Comment by mistic01** [comment]
Code Fix is working as expected - LGTM

File: timesketch/app.py [hunk 1]
```diff
@@ -145,7 +147,20 @@ def create_app(config=None, legacy_ui=False):
...
+    
+    # Returns 404 for invalid api routes
+    @app.route('/api/v1/<path:path>')
+    @login_required
+    def handle_invalid_route(path):
+        """If a request hits this route, that requested endpoint does not exist.
 
+        Returns:
+            HTTP response object (instance of jsonify)        
+        """
```
Comment (jkppr) [inline timesketch/app.py:159]:
```suggestion
        """Error handler for non-existent API routes.

        Raises:
            ApiHTTPError: Error 404 - not found        
        """
```

Updating the doc string to reflect the changes.

File: timesketch/app.py [hunk 2]
```diff
@@ -181,6 +196,8 @@ def load_user(user_id):
     # Setup CSRF protection for the whole application
     CSRFProtect(app)
 
+    
+
     return app
```
Comment (jkppr) [inline timesketch/app.py:201]:
```suggestion
    CSRFProtect(app)

    return app
```

Removing empty lines

File: timesketch/api/v1/resources_test.py [hunk 3]
```diff
@@ -43,6 +43,16 @@ def test_to_json_empty_list(self):
             },
         )
 
+class InvalidResourceTest(BaseTest):
+    """Test an Invalid Resource."""
+
+    invalid_resource_url = "api/v1/invalidresource"
+    
+    def test_invalid_endpoint(self):
```
Comment (jkppr) [inline timesketch/api/v1/resources_test.py:51]:
```suggestion
    invalid_resource_url = "api/v1/invalidresource"

    def test_invalid_endpoint(self):
```

Removing trailing whitespace making the linter happy.

**Review by jkppr — CHANGES_REQUESTED** [review]
Thanks for your PR. Please see the comments and recommended changes below.

**Review by jkppr — APPROVED** [review]
lgtm from my side. Adding @berggren for the last word.

**Pull request merged**